pandas/core/series.py

```diff
@@ -928,6 +928,17 @@ def _get_with(self, key):
                 # [slice(0, 5, None)] will break if you convert to ndarray,
                 # e.g. as requested by np.median
                 # FIXME: hack
+                if isinstance(key, list):
+                    # GH#31299
+                    warnings.warn(
+                        "Indexing with a single-item list containing a "
+                        "slice is deprecated and will raise in a future "
```

Comment:
Do we know what it will raise? Or will it depend on the value?

Reply:
Best guess is TypeError, depends on where we let it fall through